### PR TITLE
[14.0][MIG] l10n_ch_pain_base

### PR DESCRIPTION
**Pull request by SimoneVagile**
Depends on:
- [x] [https://github.com/OCA/l10n-switzerland/pull/590](https://github.com/OCA/l10n-switzerland/pull/590)

### CODE REVIEW

**Comment by SimoRubi** [comment]
Can you rebase on `14.0` since https://github.com/OCA/l10n-switzerland/pull/590 has been merged?

File: l10n_ch_pain_base/__manifest__.py
Comment (SimoRubi) [inline l10n_ch_pain_base/__manifest__.py:18]: Please include the modifications done by linters in a separated commit as explained in https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-14.0#technical-method-to-migrate-a-module-from-130-to-140-branch

File: l10n_ch_pain_base/models/account_move_line.py
Comment (SimoRubi) [inline l10n_ch_pain_base/models/account_move_line.py]: Why aren't you setting `communication_type` to `'qrr'` anymore?
Also, note that `12.0` has been recently fixed with https://github.com/OCA/l10n-switzerland/pull/607 that should be included in this PR.

**Review by SimoRubi — CHANGES_REQUESTED** [review]
Can you also check why is Travis failing? It is ✅ in `14.0`

**Comment by SimoneVagile** [comment]
> Can you also check why is Travis failing? It is white_check_mark in `14.0`

@SimoRubi `l10n_ch_base_bank` tests was failing.

I created this PR [https://github.com/OCA/l10n-switzerland/pull/609](https://github.com/OCA/l10n-switzerland/pull/609) to fix them.

File: l10n_ch_pain_base/models/account_move_line.py
Comment (SimoRubi) [inline l10n_ch_pain_base/models/account_move_line.py]: Preserve authorship and commit message of what was added by https://github.com/OCA/l10n-switzerland/pull/607, you can do that with `git cherry-pick`

File: l10n_ch_pain_base/models/account_move_line.py
Comment (SimoRubi) [inline l10n_ch_pain_base/models/account_move_line.py]: Why have you added the `_is_isr_issuer` condition here?

File: l10n_ch_pain_base/README.rst
Comment (SimoRubi) [inline l10n_ch_pain_base/README.rst]: Why is this "[UPD] README.rst" a stand-alone commit?

**Review by SimoRubi — CHANGES_REQUESTED** [review]

File: l10n_ch_pain_base/models/account_move_line.py
Comment (SimoneVagile) [reply l10n_ch_pain_base/models/account_move_line.py]: `_is_isr_issuer` checks if ISR postal code is valid.

[https://github.com/odoo/odoo/blob/14.0/addons/l10n_ch/models/res_bank.py#L32-L39](https://github.com/odoo/odoo/blob/14.0/addons/l10n_ch/models/res_bank.py#L32-L39)

File: l10n_ch_pain_base/README.rst
Comment (SimoneVagile) [reply l10n_ch_pain_base/README.rst]: Squashed

File: l10n_ch_pain_base/models/account_move_line.py
Comment (SimoRubi) [inline l10n_ch_pain_base/models/account_move_line.py]: Here and below: in `12.0` we were checking invoice's `partner_bank_id`, why are you now checking the line's `partner_bank_id`?

Note that the field `account.move.line.partner_bank_id` is not a simple related to the invoice's `partner_bank_id` but is a computed field defined in https://github.com/OCA/bank-payment/blob/7bdb286aa1a3d5beb8ee95193ed6eb4abc67ce1e/account_payment_order/models/account_move_line.py#L12

File: l10n_ch_pain_base/models/account_payment_order.py
Comment (SimoRubi) [inline l10n_ch_pain_base/models/account_payment_order.py]: Please fix the separated string into one, this is a typical `black` _mistake_

File: l10n_ch_pain_base/models/account_move_line.py
Comment (SimoRubi) [inline l10n_ch_pain_base/models/account_move_line.py]: From https://github.com/OCA/l10n-switzerland/pull/591#discussion_r675456232
> _is_isr_issuer checks if ISR postal code is valid.
>
> https://github.com/odoo/odoo/blob/14.0/addons/l10n_ch/models/res_bank.py#L32-L39

Why do you need to check if ISR postal code is valid if it was not needed in `12.0`?
If it is a new logic/fix please move it to a dedicated commit (or PR) so that the migration commit (or PR) only contains changes needed for the migration.

**Review by SimoRubi — CHANGES_REQUESTED** [review]

**Review by SimoRubi — COMMENTED** [review]
Code review: looks good to me.

Will approve when bots are 🟢; Travis is still failing but should be fixed by https://github.com/OCA/l10n-switzerland/pull/609

File: l10n_ch_pain_base/models/account_move_line.py
Comment (SimoRubi) [inline l10n_ch_pain_base/models/account_move_line.py:14]: In QRR case, the QR reference should be communicated to creditor, see https://github.com/SimoneVagile/l10n-switzerland/pull/1

**Review by SimoRubi — CHANGES_REQUESTED** [review]

File: l10n_ch_pain_base/models/account_move_line.py
Comment (SimoneVagile) [reply l10n_ch_pain_base/models/account_move_line.py:14]: Merged, thanks

**Review by stevech091 — APPROVED** [review]
Functional review ok

**Comment by yvaucher** [comment]
@SimoneVagile can you rebase? https://github.com/OCA/l10n-switzerland/pull/616 should have fixed the CI

**Comment by SimoRubi** [comment]
@SimoneVagile do I have to update my review?

**Comment by SimoneVagile** [comment]
> @SimoneVagile do I have to update my review?

@SimoRubi [https://github.com/OCA/l10n-switzerland/pull/609#issuecomment-1056992521](https://github.com/OCA/l10n-switzerland/pull/609#issuecomment-1056992521)

**Comment by hailangvn** [comment]
Good day @SimoneVagile, @SimoRubi, the travis was red because [move.partner_bank_id = bank_id](https://github.com/OCA/bank-payment/blob/14.0/account_payment_partner/models/account_move.py#L116). Re-opening [PR#609](https://github.com/OCA/l10n-switzerland/pull/609) does not have any effect because [PR#616](https://github.com/OCA/l10n-switzerland/pull/616) has already made same change.
There is proposed fix is to have [a setting to force blank partner bank](https://github.com/OCA/bank-payment/pull/890) then [use it in test](https://github.com/OCA/l10n-switzerland/pull/640). You can see that travis of [PR#641](https://github.com/OCA/l10n-switzerland/pull/641) is green thanks to the fix. The PR#641 is now having same change with this PR and can be closed.

**Comment by StephaneMangin** [comment]
Duplicate of https://github.com/OCA/l10n-switzerland/pull/591. Closing

**Comment by hailangvn** [comment]
Good day @StephaneMangin, @SimoneVagile, The [PR#640](https://github.com/OCA/l10n-switzerland/pull/640) has been updated and can be merged to fix travis. But for [PR#641](https://github.com/OCA/l10n-switzerland/pull/641), we have lost access to update it. So please help to re-open [this PR](https://github.com/OCA/l10n-switzerland/pull/591) and close [PR#641](https://github.com/OCA/l10n-switzerland/pull/641).

**Comment by i-vyshnevska** [comment]
@SimoneVagile can you pls rebase, seems the test was fixed

**Comment by michelerusti** [comment]
> @SimoneVagile can you pls rebase, seems the test was fixed

Since the owner of this PR is no longer working on Odoo, I've made #668 to continue this discussion

**Pull request closed without merging**
